frontend/src/js/components/devices/troubleshoot/terminal-wrapper.js

```diff
@@ -183,6 +202,21 @@ const TroubleshootContent = ({ device, onDownload, setSocketClosed, setUploadPat
     return close;
   }, [close, sessionState]);
 
+  useEffect(() => {
+    setIsAwaitingCheckInUpdate(false);
+  }, [device.check_in_time]);
+
+  useEffect(() => {
+    setIsAwaitingInventoryUpdate(false);
+    inventoryTimer.current = setTimeout(() => setIsAwaitingInventoryUpdate(false), TIMEOUTS.refreshLong);
```

Comment:
Wouldn't local variable (`const timerId = setTimeout(() => setIsAwaitingInventoryUpdate(false), TIMEOUTS.refreshLong);`) and  `return () => clearTimeout(timerId)` do the job? instead of using ref and 2 effects?

Reply:
Ugh, rereading this it looks like I missed a) the timer for the checkin update and setting the timers up shouldn't happen in the effect, but the click handlers 🤦 

Regular local variables aren't stable across rerenders and regular state will trigger rerenders, thus the reliance on refs - and then the effect to clear any timeouts on unmount to prevent causing errors due to state changes on an unmounted component 👌